cornflow/schemas/schema_manager.py

```diff
@@ -0,0 +1,368 @@
+
+    def dict_to_flask(self):
+        """
+        Create flask marshmallow schemas from self.dict_schema
+        
+        :return: a dict containing the flask marshmallow schemas
+        """
+        dict_params = self.schema_dict
+        result_dict = {}
+        ordered = sort_dict(dict_params)
+        tuplist = sorted(dict_params.items(), key=lambda v: ordered[v[0]])
+        for key, params in tuplist:
+            schema = ParameterSchema()
+            # this line validates the list of parameters:
+            params1 = schema.load(params, many=True)
+            result_dict[key] = gen_schema(key, params1, result_dict)
+    
+        self.flask_schemas = result_dict['DataSchema']
+        return self.flask_schemas
+    
+    def jsonschema_to_flask(self):
+        """
+        Create flask marshmallow schemas from the jsonschema
+
+        :return: a dict containing the flask marshmallow schemas
+        """
+        self.jsonschema_to_dict()
+        self.dict_to_flask()
+        return self.flask_schemas
+        
+    def load_schema(self, path):
+        """
+        Load a json schema into self.jsonschema.
+
+        :param path: the path to the json file containing the schema.json
+
+        return the jsonschema
+        """
+        self.jsonschema = self.load_json(path)
+        return self.jsonschema
+
+    def print_schema_dict(self, path):
+        """
+        Print the schema_dict in a json file.
+
+        :param path: the path where to save the dict.format
+
+        :return: nothing
+        """
+        self.save_json(self.schema_dict, path)
+    
+    def draft_schema_from(self, path, save_path=None):
+        """
+        Create a draft jsonschema from a json file of data.
+        
+        :param path: path to the json file.
+        :param save_path: path where to save the generated schema.
+        
+        :return: the generated schema.
+        """
+        
+        file = self.load_json(path)
+        
+        builder = SchemaBuilder()
+        builder.add_schema({"type": "object", "properties": {}})
+        builder.add_object(file)
+        
+        self.draft_schema = builder.to_json()
+        if save_path is not None:
+            self.save_json(self.draft_schema, save_path)
+        return self.draft_schema
+    
+    """
+    Functions used to translate jsonschema to schema_dict
+    """
+    def _create_data_schema(self, item):
+        """
+        Add a schema to self.schema_dict["DataSchema"]
+
+        :param item: (key, value) of a dict. The key cotain the name of the schema and the value containt its content.
+
+        return the schema dict.
+        """
+    
+        name = item[0]
+        content = item[1]
+    
+        schema = [dict(name=name,
+                      type=self._get_schema_name(name),
+                      many=("type" in content and content["type"] == "array"),
+                      required=True)]
+        self.schema_dict["DataSchema"] +=schema
```

Comment:
are we assuming the name "DataSchema" inside this object? does it make sense? could we just assign this outside of this object? because this name is just a name we put to the pulp instance in Cornflow.

Reply:
The last function creating the flask schema returns result_dict['DataSchema']

So the name DataSchema is not part of the final marshmallow schema. It is internal too the class and I think as good/bad as anything else.